test/python/algorithms/test_linear_solvers.py

```diff
@@ -236,7 +236,9 @@ def test_hhl(self, matrix, right_hand_side, observable):
         """Test the HHL class."""
         if isinstance(matrix, QuantumCircuit):
             num_qubits = matrix.num_state_qubits
-        elif isinstance(matrix, np.ndarray):
+        elif isinstance(matrix, (list, np.ndarray)):
```

Comment:
It would be good to add a list to the data for matrix&vector so that HHL is tested when using a list so it goes through your new conversion paths in HHL. I.e. here in the testing there should be no reason convert it to a numpy array I believe.